packages/web/src/components/track/desktop/ConnectedTrackTile.tsx

```diff
@@ -140,6 +143,8 @@ const ConnectedTrackTile = memo(
     const isTrackPlaying = isActive && isPlaying
     const isOwner = handle === userHandle
     const isArtistPick = showArtistPick && _artist_pick === trackId
+    const doesUserHaveAccess =
+      !isPremium || isOwner || !!premiumContentSignature
```

Comment:
shouldn't even need isOwner here as backend should generate signature if user is owner